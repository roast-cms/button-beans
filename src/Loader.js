// tools
import React from "react"

// styles
import styled from "styled-components"
const LoaderContainer = styled.div`
  display: inline-block;
  overflow: hidden;
  width: ${props => props.theme.size.block.spacing}em;
  height: ${props => props.theme.size.block.spacing}em;
  margin: -0.1em 0 0em -${props => props.theme.size.block.spacing / 2}em;
  svg {
    margin-top: -0.25em;
  }
`


// return
export default props => {
  return (
    <LoaderContainer {...props}>
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="28 22 20 20">
        <defs>
          <clipPath id="loader-clip">
            <rect x="20" y="0" width="44" height="40" />
          </clipPath>
        </defs>
        <path
          fill="none"
          stroke="#fff"
          strokeWidth="2"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeMiterlimit="10"
          clipPath="url(#loader-clip)"
          d="M70 38.9L70 28.3C70 25.6 67.8 23.3 65 23.3L61.6 23.3C58.9 23.3 56.6 25.5 56.6 28.3L56.6 33.9C56.6 36.6 54.4 38.9 51.6 38.9L48.2 38.9C45.5 38.9 43.2 36.7 43.2 33.9L43.2 28.3C43.2 25.6 41 23.3 38.2 23.3L35 23.3C32.3 23.3 30 25.5 30 28.3L30 33.9C30 36.6 27.8 38.9 25 38.9L21.6 38.9C18.9 38.9 16.6 36.7 16.6 33.9L16.6 28.3C16.6 25.6 14.4 23.3 11.6 23.3L8.2 23.3C5.5 23.3 3.2 25.5 3.2 28.3L3.2 33.9C3.2 36.6 1 38.9-1.8 38.9L-5 38.9C-7.7 38.9-10 36.7-10 33.9L-10 23.3"
        >
          <animateTransform
            attributeName="transform"
            type="translate"
            calcMode="linear"
            values="-20 0;7 0"
            keyTimes="0;1"
            dur="750ms"
            begin="0s"
            repeatCount="indefinite"
          />
          <animate
            attributeName="stroke-dasharray"
            calcMode="linear"
            values="0 20 23 60;0 69 22.5 10"
            keyTimes="0;1"
            dur="750ms"
            begin="0s"
            repeatCount="indefinite"
          />
        </path>
      </svg>
    </LoaderContainer>
  )
}
