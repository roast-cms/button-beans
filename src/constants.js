import { Sugar } from "@roast-cms/react-sugar-styled"

export const TEST_THEME = Sugar({
  color_brand: "rgb(189,67,54)",
  font_heading: "'Yanone Kaffeesatz', sans-serif",
  font_heading_weight: 400,
  font_body: "'Lobster Two', serif",
  size_base: 20,
  size_column_medium: 700,
  size_column_large: 900,
  size_block_spacing: .75,
  effects_border_radius: .75,
})
