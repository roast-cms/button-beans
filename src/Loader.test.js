import "jest-styled-components";

import React from "react";

import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import { Loader } from "./Loader";
import { TEST_THEME } from './constants';

test("Render button loader without crashing", () => {
  shallow(<Loader />);
});

test('Button loader matches snapshot', () => {
  const tree = renderer.create(<Loader theme={TEST_THEME} />).toJSON()
  expect(tree).toMatchSnapshot()
})
