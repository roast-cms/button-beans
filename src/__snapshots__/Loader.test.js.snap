// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button loader matches snapshot 1`] = `
.c0 {
  display: inline-block;
  overflow: hidden;
  width: 0.75em;
  height: 0.85em;
  margin: -0.1em 0 0em -0.375em;
}

<div
  className="c0"
>
  <svg
    viewBox="0 0 50 50"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
      transform="rotate(150 25 25)"
    >
      <animateTransform
        attributeName="transform"
        attributeType="xml"
        dur="0.6s"
        from="0 25 25"
        repeatCount="indefinite"
        to="360 25 25"
        type="rotate"
      />
    </path>
  </svg>
</div>
`;
